fast64_internal/sm64/__init__.py

```diff
@@ -116,11 +125,64 @@ def draw(self, context):
 		prop_split(col, context.scene, 'convertibleAddr', 'Address')
 		col.prop(context.scene, 'levelConvert')
 
+def get_legacy_export_type():
+	legacy_export_types = ('C', 'Binary', 'Insertable Binary')
+	scene = bpy.context.scene
+	if scene.get('already_derived_legacy_export_type'):
+		return scene.fast64.sm64.exportType
+	scene['already_derived_legacy_export_type'] = True
+
+	exportTypes = set()
+	exportTypes.add(legacy_export_types[scene.get('animExportType', 0)])
```

Comment:
you get a key error if you `pop` something that doesn't exist, so this method would need a try/catch and i think that could get messy. that said, it would be nice to clean out that data. so i think i'll create a loop and iterate over the 4 props and pop anyways!

___

> not a fan of adding properties directly to the scene

yeah tbh i wasn't a huge fan of this solution lol i thought of a better idea just now. I'm going to set the version to the current version after migrating.

Reply:
```python
def get_legacy_export_type():
	legacy_export_types = ('C', 'Binary', 'Insertable Binary')
	scene = bpy.context.scene

	for exportKey in ['animExportType', 'colExportType', 'DLExportType', 'geoExportType']:
		try:
			eType = scene.pop(exportKey)
			if eType and legacy_export_types[eType] != 'C':
				return legacy_export_types[eType]
		except KeyError:
			pass

	return 'C'
```
thoughts?

Reply:
just updated the PR with this change, and used a different version check